starter-code/app/src/main/java/generalassembly/yuliyakaleda/solution_code_thread_safe/MainActivity.java

```diff
@@ -60,11 +68,11 @@ private void selectImage() {
     }
 
     //TODO: Fill in the parameter types
-    private class ImageProcessingAsyncTask extends AsyncTask<> {
+    private class ImageProcessingAsyncTask extends AsyncTask<Uri,Integer,Bitmap> {
 
         //TODO: Fill in the parameter type - look at the expected type for the parameter to openInputStream()
         @Override
-        protected Bitmap doInBackground() {
+        protected Bitmap doInBackground(Uri[] params) {
```

Comment:
This might give you an error - type `Uri[]` MUST be an array, so if you pass in a single `Uri` value it will give an error because it's not an array. By comparison, the "varargs" type `Uri...` can accept either a single `Uri` value, or a bunch of them separated by commas. No need to explicitly create an array in that case.